EasyDDNS.cpp

```diff
@@ -37,14 +37,18 @@ void EasyDDNSClass::update(unsigned long ddns_update_interval){
     interval = ddns_update_interval;
 
     unsigned long currentMillis = millis(); // Calculate Elapsed Time & Trigger
+  
     if(currentMillis - previousMillis >= interval) {
         previousMillis = currentMillis;
-
-// ######## GENERATE UPDATE URL ######## //
+// ######## CHECK & UPDATE ######### //
+    if(old_ip != new_ip){
+      old_ip = new_ip;
+      // ######## GENERATE UPDATE URL ######## //
         if(ddns_choice == "duckdns"){
           update_url = "http://www.duckdns.org/update?domains="+ddns_d+"&token="+ddns_u+"&ip=";}
         else if(ddns_choice == "noip"){
-          update_url = "http://"+ddns_u+":"+ddns_p+"@dynupdate.no-ip.com/nic/update?hostname="+ddns_d+"&myip="+new_ip+"";}
+          update_url = "ht   tp://"+ddns_p+"@8.23.224.120/nic/update?hostname="+ddns_d+"&myip="+new_ip+"";}
```

Comment:
Line 50 has a rouge IP Address. **Only Authentic Domain names can be approved.**
https://www.noip.com/integrate/request

As per their website, hostname is: **dynupdate.no-ip.com**